core/src/main/scala/org/apache/spark/security/CryptoStreamUtils.scala

```diff
@@ -166,117 +158,6 @@ private[spark] object CryptoStreamUtils extends Logging {
 
   }
 
-  /**
-   * SPARK-25535. The commons-crypto library will throw InternalError if something goes
-   * wrong, and leave bad state behind in the Java wrappers, so it's not safe to use them
-   * afterwards. This wrapper detects that situation and avoids further calls into the
-   * commons-crypto code, while still allowing the underlying streams to be closed.
-   *
-   * This should be removed once CRYPTO-141 is fixed (and Spark upgrades its commons-crypto
```

Comment:
Fixed in 1.1.0 (https://commons.apache.org/proper/commons-crypto/changes-report.html). Currently used commons-crypto library in Spark is 1.1.0.

Reply:
cc @vanzin

Reply:
@vanzin left some comments at https://issues.apache.org/jira/browse/CRYPTO-141
`If I find some time I'll see whether the ASF has any infra for testing this on Win32 and MacOS, since I only have a Linux dev env available.`

Reply:
Yea, as I said that I happened to look at this code while working on related stuff. The workaround looks out-of-dated based on the status of CRYPTO-141.

Let's wait for what @vanzin thinks about the fix of CRYPTO-141.